sdk/src/main/java/com/akkaserverless/javasdk/eventsourcedentity/EventSourcedEntityBase.java

```diff
@@ -23,13 +23,17 @@
 import com.akkaserverless.javasdk.reply.MessageReply;
 
 import java.util.Collection;
+import java.util.Optional;
 import java.util.function.Function;
 
 // FIXME rename to EventSourcedEntity when the old annotation is removed
 
 /** @param <S> The type of the state for this entity. */
 public abstract class EventSourcedEntityBase<S> {
 
+  private Optional<CommandContext> commandContext = Optional.empty();
+  private Optional<EventContext> eventContext = Optional.empty();
```

Comment:
We toggle these between empty and actually containing a context for the respective calls, we don't expect that they will be used in the general case anymore since they only contain access to metadata etc. which usually isn't needed.